lib/breadcrumbs.ts

```diff
@@ -9,33 +9,50 @@ interface CrumbData {
   title: string;
 }
 
-export function getCrumbStringFromIssueData(issue: Pick<IssueData, 'html_url' | 'title'>): string {
-  return `${convertGithubUrlToShorthand(new URL(issue.html_url))}@@${issue.title}`
+export function getCrumbStringFromIssueData({ html_url, title }: Pick<IssueData, 'html_url' | 'title'>): string {
+  return JSON.stringify(getCrumbDataArrayFromIssueData({ html_url, title }));
+}
+export function getCrumbDataArrayFromIssueData({ html_url, title }: Pick<IssueData, 'html_url' | 'title'>): [string, string] {
+  return [convertGithubUrlToShorthand(new URL(html_url)), title];
+}
+
+export function convertCrumbDataArraysToCrumbDataString(crumbDataArrays: [string, string][]): string {
+  return JSON.stringify([...crumbDataArrays]);
+}
+
+function mapCrumbArrayToCrumbData(crumbItem, index, crumbArray) {
+  const [shortGithubId, title] = crumbItem;
+  const { owner, repo, issue_number } = paramsFromUrl(getValidUrlFromInput(shortGithubId).toString());
+  const url = new URL(`/roadmap/github.com/${owner}/${repo}/issues/${issue_number}`, window.location.origin);
+  const crumbParents = crumbArray.slice(0, index);
+
+  if (crumbParents.length > 0) {
+    const crumbsForUrl = convertCrumbDataArraysToCrumbDataString([ ...crumbParents]);
+
+    url.searchParams.append('crumbs', crumbsForUrl)
+  }
+
+  return {
+    url,
+    title,
+  };
 }
 
 /**
  *
- * @param crumbs A string of decoded crumbs. Example: `owner/repo#1@@Issue 1,owner/repo#2@@Issue 2`
+ * @param crumbs A JSON.stringified representation of crumbs like that returned by convertCrumbDataArraysToCrumbDataString
  * @param viewMode the current view mode, to ensure links stay in the current view
  * @returns
  */
 export function getCrumbDataFromCrumbString(crumbs: string, viewMode: ViewMode): CrumbData[] {
-  const crumbItems = crumbs.split(',');
-  const crumbData: CrumbData[] = [];
-  crumbItems.forEach((crumb, index) => {
-    const crumbParts = crumb.split('@@');
-    const { owner, repo, issue_number } = paramsFromUrl(getValidUrlFromInput(crumbParts[0]).toString());
-    const url = new URL(`/roadmap/github.com/${owner}/${repo}/issues/${issue_number}#${viewMode}`, window.location.origin);
-    const crumbsForUrl = crumbItems.slice(0, index).join(',');
-
-    if (crumbsForUrl.length > 0) {
-      url.searchParams.append('crumbs', crumbsForUrl)
-    }
-
-    crumbData.push({
-      url: url.toString(),
-      title: crumbParts[1],
-    });
+  const crumbItems = JSON.parse(crumbs) as [string, string][];
+
+  const crumbData: CrumbData[] = crumbItems.map(mapCrumbArrayToCrumbData).map((crumb) => {
```

Comment:
looping over them twice here, but the size of the breadcrumbs should never be larger than 10.. and that would be extreme.